### PR TITLE
Replace nose with py.test

### LINKED ISSUE #991: Support pytest as test runner
Actually tests basically work, the only issue is with the attr marker from nose. Branch [feature/pytest][1] implements a [meta decorator][2] (yes, really) because nose and pytest implementations of test tagging are similar, but different enough to be incompatible.

Trying to import pytest first is a bit weird, but I didn't find any way to check which test runner is active. This still works with nose even if pytest is present, so we don't break how it worked previously.

[1]: https://github.com/ged-lab/khmer/compare/feature/pytest
[2]: https://github.com/ged-lab/khmer/compare/feature/pytest#diff-356b0cce172f19b23235de06daf67197R19

### PR DESCRIPTION
**Pull request by luizirber**
closes #991 #1354

- [x] Is it mergeable?
- [x] Did it pass the tests?
- [x] If it introduces new functionality in scripts/ is it tested?
  Check for code coverage with `make clean diff-cover`
- [x] Is it well formatted? Look at `make pep8`, `make diff_pylint_report`,
  `make cppcheck`, and `make doc` output. Use `make format` and manual
  fixing as needed.
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Is it documented in the ChangeLog?
  http://en.wikipedia.org/wiki/Changelog#Format
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?
- [x] Do the changes respect streaming IO? (Are they
  tested for streaming IO?)
- [x] Is the Copyright year up to date?

### CODE REVIEW

**Comment by luizirber** [comment]
Documentation TODO:
- [ ] [doc/user/install.rst][1]
- [ ] [doc/dev/a-quick-guide-to-testing.rst][2]
- [ ] [doc/dev/release.rst][3]
- [ ] [doc/dev/getting-started.rst][4]
- [ ] Comment on [setup.py][5]

[1]: https://github.com/dib-lab/khmer/blob/b43907695b86f593487d06a38d702ee70988a6f9/doc/user/install.rst
[2]: https://github.com/dib-lab/khmer/blob/b43907695b86f593487d06a38d702ee70988a6f9/doc/dev/a-quick-guide-to-testing.rst
[3]: https://github.com/dib-lab/khmer/blob/b43907695b86f593487d06a38d702ee70988a6f9/doc/dev/release.rst
[4]: https://github.com/dib-lab/khmer/blob/b43907695b86f593487d06a38d702ee70988a6f9/doc/dev/getting-started.rst
[5]: https://github.com/dib-lab/khmer/blob/b43907695b86f593487d06a38d702ee70988a6f9/setup.py#L294

File: doc/dev/a-quick-guide-to-testing.rst [hunk 1]
```diff
@@ -119,10 +119,10 @@ scratch, or copy an existing one.  (We recommend the latter.)
 
 To write a new one, you'll need to know how to write tests. For
 getting an idea of the syntax, read this `introductory guide
-<http://ivory.idyll.org/articles/nose-intro.html>`__ and the `official
-documentation
-<https://nose.readthedocs.org/en/latest/writing_tests.html>`__.  Then
-find the right file in ``tests/*.py`` and add your test!
+<http://ivory.idyll.org/articles/nose-intro.html>`__ and the `writing tests
+documentation from Astropy
+<http://docs.astropy.org/en/v1.1/development/testguide.html#writing-tests>`__.
```
Comment (luizirber) [inline doc/dev/a-quick-guide-to-testing.rst]: I added the astropy docs as example, they are pretty good.

File: doc/user/install.rst [hunk 2]
```diff
@@ -94,17 +94,3 @@ Installing khmer inside the virtualenv
...
-After installing you can run the embedded test suite. If you are running an
-OSX system you should also add `!linux` to prevent linux-specific tests from
-failing.::
-
-      nosetests khmer --attr '!known_failing,!huge'
-
-If the nosetests binary isn't installed then::
-
-      pip2 install khmer[tests]
-      nosetests khmer --attr '!known_failing,!huge'
```
Comment (luizirber) [inline doc/user/install.rst]: I didn't find an easy way to run the installed package tests, so I removed this section. If anyone has better suggestions...

**Comment by luizirber** [comment]
Still need to update the release docs, but they probably need a larger review, considering we don't do a release for quite some time

**Comment by luizirber** [comment]
Otherwise, ready for review @ctb @camillescott

File: doc/user/install.rst (hunk 2 above)
Comment (ctb) [inline doc/user/install.rst]: We'll have to look at this again around release-time, I think.

**Comment by ctb** [comment]
Nice work!

File: doc/user/install.rst (hunk 2 above)
Comment (standage) [inline doc/user/install.rst]: This just worked for me.

```bash
deactivate
virtualenv ../temp-env
source ../temp-env/bin/activate
make install-dep
make clean
make install
cd ..
py.test --pyargs khmer.tests -m 'not known_failing and not jenkins and not huge and not linux'
```

I should clarify: by "this worked" I mean "the tests ran". Nine sandbox tests failed, as well as the perennial "C version == Py version" test. But that could've been me.

Comment (standage) [inline doc/user/install.rst]: Hah, looks like I was a minute too late. I'll do another PR with this.

**Pull request merged**